internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -329,20 +342,89 @@ func (c *cmd) createEdgeXACLTokenRoles(bootstrapACLTokenID, secretstoreToken str
 		}
 
 		// create roles based on the service keys as the role names
-		edgexACLTokenRole := NewRegistryRole(roleName, ClientType, []Policy{
+		edgexACLTokenRole := types.NewRegistryRole(roleName, types.ClientType, []types.Policy{
 			*edgexServicePolicy,
 			// localUse set to false as some EdgeX services may be running in a different node
 		}, false)
 
-		// fail all if any one of the role creation failed
-		if err := c.createRole(secretstoreToken, edgexACLTokenRole); err != nil {
+		clientConfig := types.SecretConfig{
+			Type:     secrets.Vault,
+			Host:     c.configuration.SecretStore.Host,
+			Port:     c.configuration.SecretStore.Port,
+			Protocol: c.configuration.SecretStore.Protocol,
+		}
+		client, err := secrets.NewSecretStoreClient(clientConfig, c.loggingClient, c.client)
+		if err != nil {
+			return fmt.Errorf("failed to create SecretStoreClient: %s", err.Error())
+		}
+
+		if err := client.CreateRole(secretstoreToken, edgexACLTokenRole); err != nil {
 			return fmt.Errorf("failed to create edgex role: %v", err)
 		}
 	}
 
 	return nil
 }
 
+// configureConsulAccess is to enable the Consul config access to the SecretStore via consul/config/access API
+// see the reference: https://www.vaultproject.io/api-docs/secret/consul#configure-access
+func (c *cmd) configureConsulAccess(secretStoreToken string, bootstrapACLToken string) error {
```

Comment:
i thought this func already moved to go.mod.secrets?

Reply:
My mistake, will remove